utHelper.py

```diff
@@ -348,16 +311,23 @@ def downloadToDevice(self, urls: list, target_directory: str, device: str="dut"
         self.log.debug("urls( url:'{}' )".format(urls))
         self.log.debug("target_directory( target_directory:'{}' )".format(target_directory))
 
-        for url in urls:
-            # Download a file into the workspace on the host
-            # Then copy the file to target
-            file_name = os.path.basename(url)
-            if hasattr(self, 'outboundClient'):
-                self.outboundClient.downloadFile(url)
-                workspace_directory = self.outboundClient.workspaceDirectory
-                self.copyFileFromHost(os.path.join(workspace_directory, file_name), target_directory, targetDevice=device)
-            else:
-                self.log.error("outboundClient not present")
+        utils = utBaseUtils()
```

Comment:
you should start the helper class in the init of utHelper, there should only be one of them, not launched for every usecase during the functions.

```python
self.utils = utBaseUtils()
```

Then it's available always in this class.

Reply:
done